lib/client.ex

```diff
@@ -17,6 +17,14 @@ defmodule Mailgun.Client do
   end
 
   def send_email(conf, email) do
+    if conf[:mode] == :testing do
```

Comment:
Let's change this to do_send_email(conf[:mode], ...

then we can just pattern match mulitple clauses of `do_send_email(:test, ..` and skip the if else dance